app/main.py

```diff
@@ -2,0 +1,34 @@
+from __future__ import annotations
+
+
+class Animal:
+    alive = []
+
+    def __init__(self,
+                 name: str,
+                 heals: int = 100,
+                 hidden: bool = False) -> None:
+        self.name = name
+        self.health = heals
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+
+
+class Carnivore(Animal):
+    @classmethod
+    def bite(cls, herbivore: Animal) -> None:
+        if isinstance(herbivore, Herbivore) and not herbivore.hidden:
+            herbivore.health -= 50
+
+        if herbivore.health <= 0:
+            cls.alive.remove(herbivore)
```

Comment:
Code Efficiency: This method could be a static method since it doesn't use any instance attributes. Refer to point 3 in the checklist.